Sources/SwiftParser/Lexer/LexemeSequence.swift

```diff
@@ -29,7 +29,7 @@ extension Lexer {
     ///
     /// The memory footpring of not freeing past lexer states is neglible. It's
     /// usually less than 0.1% of the memory allocated by the syntax arena.
-    var lexerStateAllocator = BumpPtrAllocator(slabSize: 256)
+    var lexerStateAllocator = BumpPtrAllocator(slabSize: 512)
```

Comment:
Mainly a gut feeling, I did benchmark, and there was a slight improvement, but it was pretty close to margin of error tbh. I'm okay not changing this. `RegexLiteralLexemes` is 7 words, so this accommodates 8 regex literals before needing another slab (including two words for the state itself).

Reply:
I’m happy with either value. I was just interested in what made you change it. I'll trust your judgment. Just remember that most code doesn’t contain regex literals, so we shouldn’t be optimizing too much for regex literals being present.

Reply:
That's fair, though this is also bearing in mind that we're now also burning 2 words every time we see `try?`/`try!`. So even without regex literals, a slab size of 256 only accommodates 16 string literals/`try?`/`try!`. That's probably fine, but 32 feels a bit better. I really don't feel too strongly about this though.

Reply:
I don’t think we need to allocate 2 words for every `try?`/`try!` because we can store a single state inline with the `StateStack` without having to do any allocation on the bump allocator.

Reply:
The same goes for the regex state since you don’t nest regex states. So I wouldn’t expect us to actually hit the bump allocator for regexes at all.

Reply:
> I don’t think we need to allocate 2 words for every `try?`/`try!` because we can store a single state inline with the StateStack without having to do any allocation on the bump allocator.

Hmm I guess we could do that if we know we're already in the normal lexing state, but we wouldn't be able to do that if we're in an interpolation, right?

> The same goes for the regex state since you don’t nest regex states. So I wouldn’t expect us to actually hit the bump allocator for regexes at all.

Same as above, also we'd still need to allocate the lexemes struct on the allocator as we can't store that inline.

Reply:
Ahhhh I realize I misunderstood the state transition code, pushing a state only allocates when there are n > 1 states, so it doesn't affect `try?`/`try!`/regex literals. We still need to allocate the lexemes themselves on the bump pointer allocator, but I think this change probably is less compelling with that being the only thing to consider. I'll revert it.

Reply:
I was just writing a comment saying exactly that.